File: ase/misc/ntloop.js

var Math = Java['java.lang.Math'];
var Thread = Native.Util.Thread;

function Worker(id) {
  return function() {
    r = Math.abs(-99);
    // print(r);
  }
}

var loop = 10000;
for (var i = 0; i < loop; ++i) {
  var thr = Thread(Worker(i));
  try {
    thr.start();
  } finally {
    thr.join();
  }
}

File: ase/misc/ntloop.lua

local Math = Java['java.lang.Math']
local Thread = Native.Util.Thread

function Worker(id)
  return function()
    local r = Math.abs(-99)
    -- print(r)
  end
end

local loop = 100000
for i = 0, loop do
  local thr = Thread(Worker(i))
  thr.start()
  thr.join()
end

